### PR TITLE
fix(magicStringNotes): EVAL-344 fix double magic strings for same words

### PR DESCRIPTION
**Pull request by simonchaval**
## Describe your changes
Move existant constant in the right package and file.
Fix duplicate constant for same string and replace this constant by a magic string.
Refactor rename VSCO_SCHEMA in VIESCO_SCHEMA.
## Checklist tests
- Be sure to test every changes made by these magic string and refactor.
## Issue ticket number and link
EVAL-344 : https://entsupport.gdapublic.fr/browse/EVAL-344
## Checklist before requesting a review (magic string, indentation, comment/documentation...)
- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/competences/service/impl/DefaultLSUService.java [hunk 1]
```diff
@@ -270,7 +270,7 @@ public void remUnheededStudents(JsonArray idsStudents, Long idPeriode, String id
                                     final Handler<Either<String, JsonArray>> handler){
 
         SqlStatementsBuilder statements = new SqlStatementsBuilder();
-        String query = " DELETE FROM " + Competences.EVAL_SCHEMA + "." + LSU_UNHEEDED_STUDENTS_TABLE
+        String query = " DELETE FROM " + Competences.COMPETENCES_SCHEMA + "." + LSU_UNHEEDED_STUDENTS_TABLE
                 + " WHERE id_eleve =? AND  id_classe = ? "
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultLSUService.java:274]: supprime Competences

File: src/main/java/fr/openent/competences/service/impl/DefaultLSUService.java [hunk 2]
```diff
@@ -282,7 +282,7 @@ public void remUnheededStudents(JsonArray idsStudents, Long idPeriode, String id
     public void getUnheededStudents(JsonArray idPeriodes, JsonArray idClasses,
                                     final Handler<Either<String, JsonArray>> handler){
         StringBuilder query = new StringBuilder().append(" SELECT * ").append(" FROM ")
-                .append(Competences.EVAL_SCHEMA).append(".").append(LSU_UNHEEDED_STUDENTS_TABLE)
+                .append(Competences.COMPETENCES_SCHEMA).append(".").append(LSU_UNHEEDED_STUDENTS_TABLE)
                 .append(" WHERE id_classe IN ").append(Sql.listPrepared(idClasses.getList()))
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultLSUService.java:286]: supprime Competences

File: src/main/java/fr/openent/competences/service/impl/DefaultNoteService.java [hunk 3]
```diff
@@ -334,7 +334,7 @@ public void getNoteElevePeriode(String userId, String etablissementId, JsonArray
                 .append(COMPETENCES_SCHEMA).append(".rel_devoirs_groupes ")
                 .append(" ON rel_devoirs_groupes.id_devoir = devoirs.id AND rel_devoirs_groupes.id_groupe IN ")
                 .append(Sql.listPrepared(idsClass.getList())).append(" LEFT JOIN ")
-                .append(Competences.VSCO_SCHEMA).append(".services ")
+                .append(Competences.VIESCO_SCHEMA).append(".services ")
                 .append(" ON (rel_devoirs_groupes.id_groupe = services.id_groupe ")
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultNoteService.java:338]: supprime Competences

File: src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java [hunk 4]
```diff
@@ -93,7 +93,7 @@ private void conditionsToDoTransition(String idStructureATraiter, Handler<Either
 
         String queryDevoir = "SELECT id FROM " + Competences.COMPETENCES_SCHEMA + ".devoirs WHERE id_etablissement= ? AND owner !='id-user-transition-annee' ";
 
-        String queryPeriode = "SELECT id FROM " + Competences.VSCO_SCHEMA + ".periode WHERE id_etablissement = ? ";
+        String queryPeriode = "SELECT id FROM " + Competences.VIESCO_SCHEMA + ".periode WHERE id_etablissement = ? ";
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java:97]: supprime Competences

File: src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java [hunk 5]
```diff
@@ -201,10 +201,10 @@ private Handler<Either<String, JsonArray>> handlerGetInfosClasses(Map<String, Li
                 executeTransitionForStructure(classeIdsEleves, vListIdsGroupesATraiter, vMapGroupesATraiter,
                         idStructureATraiter, finalHandler);
             }else{
-                log.warn("transition année :  erreur lors de la récupération des classes dans la table " + Competences.VSCO_SCHEMA + ".periode :" +
+                log.warn("transition année :  erreur lors de la récupération des classes dans la table " + Competences.VIESCO_SCHEMA + ".periode :" +
                         " id Etablissement : " + idStructureATraiter);
                 finalHandler.handle(new Either.Left<>(
-                        "transition année :  erreur lors de la récupération des classes dans la table " + Competences.VSCO_SCHEMA + ".periode :" +
+                        "transition année :  erreur lors de la récupération des classes dans la table " + Competences.VIESCO_SCHEMA + ".periode :" +
                                 " id Etablissement : " + idStructureATraiter));
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java:208]: supprime Competences

File: src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java [hunk 6]
```diff
@@ -335,7 +335,7 @@ private void manageDevoirsAndCompetences( String idStructureATraiter, Map<String
             String queryInsertDevoir = "WITH temp_periode AS ( " +
                     "SELECT type.id as id_type, MAX(periode.id_type) as max_periode_id_type, MAX(periode.date_fin_saisie) as max_periode_date_fin_saisie," +
                     "periode.id_classe as periode_id_classe " +
-                    "FROM " + Competences.COMPETENCES_SCHEMA + ".type , " + Competences.VSCO_SCHEMA + ".periode WHERE type.default_type = ? AND type.id_etablissement = ? AND periode.id_etablissement = ? " +
+                    "FROM " + Competences.COMPETENCES_SCHEMA + ".type , " + Competences.VIESCO_SCHEMA + ".periode WHERE type.default_type = ? AND type.id_etablissement = ? AND periode.id_etablissement = ? " +
                     "GROUP BY periode.id_etablissement,type.id, periode.id_classe ) " +
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java:339]: supprime Competences

File: src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java [hunk 7]
```diff
@@ -511,10 +511,10 @@ public void clearTablePostTransition(Handler<Either<String, JsonArray>> handler)
...
                 Competences.COMPETENCES_SCHEMA + "." + Competences.STUDENT_APPRECIATION_DIGITAL_SKILLS + ", " +
                 Competences.COMPETENCES_SCHEMA + "." + Competences.STUDENT_DIGITAL_SKILLS_TABLE + ", " +
-                Competences.VSCO_SCHEMA + "." + Competences.VSCO_ABSENCES_ET_RETARDS + ", " +
-                Competences.VSCO_SCHEMA + "." + Competences.VSCO_PERIODE + ", " +
-                Competences.VSCO_SCHEMA + "." + Competences.VSCO_MULTI_TEACHING + ", " +
-                Competences.VSCO_SCHEMA + "." + Competences.VSCO_SERVICES_TABLE;
+                Competences.VIESCO_SCHEMA + "." + Competences.VSCO_ABSENCES_ET_RETARDS + ", " +
+                Competences.VIESCO_SCHEMA + "." + Competences.VSCO_PERIODE + ", " +
+                Competences.VIESCO_SCHEMA + "." + Competences.VSCO_MULTI_TEACHING + ", " +
+                Competences.VIESCO_SCHEMA + "." + Competences.VSCO_SERVICES_TABLE;
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java:518]: supprime Competences

File: src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java [hunk 8]
```diff
@@ -566,12 +566,12 @@ public JsonArray createStatements(final String currentYear){
                 .append("SELECT function_clone_schema_with_sequences(?::text, ?::text, TRUE)");
 
         statements.add(new JsonObject()
-                .put("statement", "ALTER SCHEMA " + Competences.VSCO_SCHEMA + " RENAME TO " + Competences.VSCO_SCHEMA + "_" + currentYear)
+                .put("statement", "ALTER SCHEMA " + Competences.VIESCO_SCHEMA + " RENAME TO " + Competences.VIESCO_SCHEMA + "_" + currentYear)
                 .put("values", new fr.wseduc.webutils.collections.JsonArray())
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java:570]: supprime Competences

File: src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java [hunk 9]
```diff
@@ -749,7 +749,7 @@ public void supprimerSousMatiereNonRattaches(JsonArray matieres, final Handler<E
                 values.add(id);
         }
 
-        query.append("DELETE FROM ").append(Competences.VSCO_SCHEMA).append(".").append(Competences.VSCO_SOUS_MATIERE_TABLE)
+        query.append("DELETE FROM ").append(Competences.VIESCO_SCHEMA).append(".").append(Competences.VSCO_SOUS_MATIERE_TABLE)
                 .append(" WHERE id_matiere NOT IN ").append(Sql.listPrepared(matieres.getList()));
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultTransitionService.java:753]: supprime Competences

File: src/main/java/fr/openent/competences/service/impl/DefaultUtilsService.java [hunk 10]
```diff
@@ -116,7 +116,7 @@ public void getTitulaires(String psIdRemplacant, String psIdEtablissement, Handl
         JsonArray values = new fr.wseduc.webutils.collections.JsonArray();
 
         query.append("SELECT DISTINCT main_teacher_id ")
-                .append("FROM " + Competences.VSCO_SCHEMA + ".multi_teaching ")
+                .append("FROM " + Competences.VIESCO_SCHEMA + ".multi_teaching ")
                 .append("WHERE second_teacher_id = ? ")
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultUtilsService.java:120]: supprime Competences

File: src/main/java/fr/openent/competences/service/impl/DefaultUtilsService.java [hunk 11]
```diff
@@ -315,14 +315,14 @@ public void getInfoEleveBackup(String id, Handler<Either<String, JsonObject>> re
                 // Selection
                 .append(" (SELECT personnes_supp.id_user AS \"idEleve\", MAX(delete_date) AS \"deleteDate\", ")
                 .append(" string_agg(DISTINCT rel_groupes_personne_supp.id_groupe, ',') AS \"idGroupes\" ")
-                .append(" FROM " + Competences.VSCO_SCHEMA + ".personnes_supp, viesco.rel_groupes_personne_supp ")
+                .append(" FROM " + Competences.VIESCO_SCHEMA + ".personnes_supp, viesco.rel_groupes_personne_supp ")
                 .append(" WHERE personnes_supp.id = rel_groupes_personne_supp.id ")
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/competences/service/impl/DefaultUtilsService.java:319]: supprime Competences

**Comment by lapeyronnie** [comment]
Vérifie que Competences.COMPETENCES_SCHEMA n'est pas encore utilisé

**Comment by simonchaval** [comment]
Wrong method, now gonna use file the existant "Field.java" with magic string key to update in the continuity of the project.
cf : [This PR](https://github.com/OPEN-ENT-NG/competences/pull/127)

**Pull request closed without merging**